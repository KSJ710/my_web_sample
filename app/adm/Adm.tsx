type Props = {
  categories: JSX.Element
  posts: JSX.Element
}

export default function Adm(props: Props): JSX.Element {
  // const [selectedTag, setSelectedTag] = useState(null)

  return (
    <div className="mt-[-32px] p-4">
      {props.categories}
      {props.posts}
    </div>
  )
}
