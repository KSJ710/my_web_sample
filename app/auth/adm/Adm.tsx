export default function Adm(props: any): JSX.Element {
  // const [selectedTag, setSelectedTag] = useState(null)

  return (
    <div className="mt-[-32px] p-4">
      {props.categories}
      {props.posts}
    </div>
  )
}
