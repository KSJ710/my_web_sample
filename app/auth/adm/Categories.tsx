import { prisma } from 'lib/prisma'

export default async function Categories(): Promise<JSX.Element> {
  const categories: MTagCategory[] = await prisma.mTagCategory.findMany({})

  const categoryList = categories.map((category: MTagCategory) => (
    <span
      key={category.id}
      className="inline-block border-2 border-solid border-main bg-accent px-2 py-1 text-xs text-white"
    >
      {replaceOverWordCount(category.name)}
    </span>
  ))

  return <div className="mb-[24px] flex gap-2 overflow-scroll whitespace-nowrap">{categoryList}</div>
}

function replaceOverWordCount(text: string): string {
  const wordCount: number = text.length
  const wordLimit: number = 16

  if (wordCount >= wordLimit) {
    let targetWord = text.slice(wordLimit, wordCount + 1)
    return text.replace(targetWord, '...')
  }

  return text
}
