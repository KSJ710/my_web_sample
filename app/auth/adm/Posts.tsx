/* eslint-disable tailwindcss/no-custom-classname */
import { prisma } from 'lib/prisma'
import Image from 'next/image'
import dummy from 'public/images/dummy.png'

export default async function Posts(): Promise<JSX.Element> {
  const posts: Post[] = await getPosts()

  const post_list = posts.map((post: Post) => (
    <article key={post.id} className="">
      <time className="font_yellow_underline_green mb-[8px] text-lg">{dateToYmd(post.updatedAt)}</time>
      <Image src={dummy} alt="Picture of the author" className="py-1" />
      <h2 className="font_yellow_underline_green text-2xl">{post.title}</h2>
    </article>
  ))

  return (
    <article>
      <div>
        <h1 className="font_yellow_underline_green mb-[16px] font-bold">記事一覧</h1>
        <div className="grid gap-4 sm:grid-cols-2 lg:grid-cols-3 2xl:grid-cols-4">{post_list}</div>
      </div>
    </article>
  )
}

async function getPosts(): Promise<Post[]> {
  return await prisma.post.findMany({})
}

// stringDate→yyyy/mm/dd
export function dateToYmd(date: Date) {
  var year = date.getFullYear()
  var month = date.getMonth() + 1
  var day = date.getDate()

  return ` ${year} / ${month} / ${day}`
}
