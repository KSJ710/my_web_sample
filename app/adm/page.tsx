import { Suspense } from 'react'
import Adm from './Adm'
import Category from './Category'
import Post from './Post'

export default function Page(): JSX.Element {
  return (
    <Adm
      categories={
        <Suspense fallback={<p>Loading...1</p>}>
          {/* @ts-expect-error Server Component */}
          <Category />
        </Suspense>
      }
      posts={
        <Suspense fallback={<p>Loading...2</p>}>
          {/* @ts-expect-error Server Component */}
          <Post />
        </Suspense>
      }
    />
  )
}

Page.props = {
  authStatus: 'authenticated'
}
