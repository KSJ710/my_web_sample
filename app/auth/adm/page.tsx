import { Suspense } from 'react'
import Adm from './Adm'
import Categories from 'components/fetches/Categories'
import Posts from 'components/fetches/Posts'

export default function Page() {
  return (
    <Adm
      categories={
        <Suspense fallback={<p>Loading...1</p>}>
          <Categories />
        </Suspense>
      }
      posts={
        <Suspense fallback={<p>Loading...2</p>}>
          <Posts />
        </Suspense>
      }
    />
  )
}

Page.props = {
  authStatus: 'authenticated'
}
