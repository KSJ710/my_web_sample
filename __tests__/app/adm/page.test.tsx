// 非同期コンポーネントのレンダリング出来ず
import { replaceOverWordCount } from 'app/adm/Category'
import { act } from 'react-dom/test-utils'

beforeEach(() => {})

afterEach(() => {})

describe('auth/adm replaceOverWordCount', () => {
  it('fn short word', () => {
    const result = replaceOverWordCount('123456789012345')
    expect(result).toBe('123456789012345')
  })
  it('fn long word', () => {
    const result = replaceOverWordCount('1234567890123456')
    expect(result).toBe('123456789012345...')
  })
})
