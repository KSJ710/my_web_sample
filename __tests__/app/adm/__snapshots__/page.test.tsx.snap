// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth/adm render page unchanged 1`] = `<div />`;
